sources/src/citygml/citygmlfactory.cpp

```diff
@@ -186,4 +188,29 @@ namespace citygml {
     {
         m_appearanceManager->addAppearanceTarget(obj);
     }
+
+    const std::string CityGMLFactory::getCodeValue(const std::string codeSpace, int id, const std::string gmlpath)
+    {
+        const std::string codeSpacePath = std::filesystem::absolute(gmlpath + "/../" + codeSpace).string();
+
+        if (m_codeLists->find(codeSpacePath) == m_codeLists->end()) {
+            if (!std::filesystem::exists(codeSpacePath)) {
+                CITYGML_LOG_ERROR(m_logger, "Can not find codelist file " << codeSpacePath);
+                throw std::runtime_error("Unexpected Error occurred while parsing xml file.");
+            }
+            CITYGML_LOG_INFO(m_logger, "parsing " << codeSpacePath);
+
+            CodeListHandlerXerces handler;
+            CodeList code_list = handler.getCodeList(codeSpacePath);
+            m_codeLists->emplace(codeSpacePath, code_list);
+        }
+        
+        std::string codeValue = "None";
```

Comment:
デフォルト値は空(`""`)がいいと思います。
```suggestion
        std::string codeValue;
```